cloud/ec2.go

```diff
@@ -225,7 +225,8 @@ func (m *ec2Manager) spawnOnDemandHost(ctx context.Context, h *host.Host, ec2Set
 		ec2Settings.UserData = expanded
 	}
 
-	userData, err := bootstrapUserData(ctx, evergreen.GetEnvironment(), h, ec2Settings.UserData)
+	env := evergreen.GetEnvironment()
```

Comment:
can we make it so that we pass an environment into the manager constructor rather than calling GetEnvironmnet? 

It's fine if this is in a ticket?